src/OpenTelemetry.Instrumentation.Http/README.md

```diff
@@ -55,18 +55,140 @@ For an ASP.NET application, adding instrumentation is typically done in the
 `Global.asax.cs`. Refer to documentation for [OpenTelemetry.Instrumentation.AspNet](../OpenTelemetry.Instrumentation.AspNet/README.md).
 
 **`AddHttpClientInstrumentation` vs. `AddHttpWebRequestInstrumentation`**
-For .NET Framework, `AddHttpWebRequestInstrumentation` can be used to only add
-instrumentation for `HttpWebRequest`. Using `AddHttpClientInstrumentation` in a
-.NET Framework application will add instrumentation for both `HttpClient` and
-`HttpWebRequest`.
+
+For .NET Framework, two extensions methods exist:
+`AddHttpClientInstrumentation` and `AddHttpWebRequestInstrumentation`.
+.NET Framework's implementation of HttpClient uses HttpWebRequest internally,
+so using one extension method over the other allows for configuring options
+independently for HttpClient instrumentation vs. HttpWebRequest
+instrumentation.
```

Comment:
Based on our conversation regarding HttpClient and HttpWebRequest, I took a stab at better wording for this section, though I'm not convinced that it is accurate because our conversation has made me a bit confused by this code

https://github.com/open-telemetry/opentelemetry-dotnet/blob/ce0bf3f7238d9e7ce9c79772fb5298445827c51d/src/OpenTelemetry.Instrumentation.Http/TracerProviderBuilderExtensions.cs#L30-L42

This seems to suggest the ability to set options for HttpClient _and_ HttpWebRequest, but if I'm understanding things correctly, the options passed to this method for HttpClient effectively remain unused. Right?

Reply:
mm. The TracerProviderBuilderExtensions code seems bit puzzling to me as well. Let me take another close look and get back to you.

Reply:
In .NET Framework, HttpClient uses HttpWebRequest. So really, only AddHttpWebRequestInstrumentation will be active. But if you use a .NET Standard package that pulls in .NET Standard HttpClient, what will happen? Could .NET Core bits get in there? Not totally sure about that but I think that's why both are there.

Reply:
Hmm. I'm pretty sure the guidance is to not consume HttpClient as a nuget package for .net framework. While it's technically possible it often leads to problems. I feel like I've read more "official" guidance on this somewhere but all I'm finding offhand is a tweet from Immo Landsworth https://twitter.com/terrajobst/status/997262020108926976?s=20.

I guess my thought is if there's a general consensus that this isn't best practice then should we seek to support it?

Reply:
should we just remove AddHttpWebRequest to avoid confusion, and have a single method `AddHttpClientInstrumentation` for both .NET core and .NET framework?
And in the documentation for filter, state that, In .NET Core, HttpRequestMessage and .NET Framework, HttpWebRequest is the type passed to Filter?

Reply:
For now, I just removed this section distinguishing between the two add methods. I actually didn't realize until just now that `AddHttpWebRequestInstrumentation` is an internal method - sorry 🤦 - so in effect this was already true:

> should we just remove AddHttpWebRequest to avoid confusion, and have a single method AddHttpClientInstrumentation for both .NET core and .NET framework?

Though, I am still curious about the behavior of things when the NuGet package is used in the context of .NET Framework. I'll spend a little time experimenting with this, because I feel if we can simplify and remove the `configureHttpClientInstrumentationOptions` for .NET Framework, that would be kinda nice.